### ISSUE TITLE
Add support for CRAM

### ISSUE THREAD

**Issue by MaxUlysse** [SciLifeLab/Sarek#511]

**Comment by MaxUlysse** [comment]
It takes around twice less space, and it's getting used by more and more tools.
So it might be a good idea to look into it.

**Comment by marcelm** [comment]
Thanks for opening the issue :-). Many tools support CRAM already, in particular GATK and samtools. I’ve also just found out that Manta supports CRAM natively.

Here’s a list of processes in CAW that currently process BAM files as input or output. I’m listing whether the tool that is used also supports CRAM.

* MapReads (samtools sort): Yes
* MergeBams (samtools merge): Yes
* MarkDuplicates (Picard): Yes
* RealignerTargetCreator (GATK): Yes
* IndelRealigner (GATK): Yes
* CreateRecalibrationTable (GATK): Yes
* RecalibrateBam (GATK): Yes
* RunSamtoolsStats (samtools stats): Yes
* RunBamQC (qualimap): [Probably no](https://bitbucket.org/kokonech/qualimap/issues/34/)
* RunHaplotypecaller (GATK): Yes
* RunGenotypeGVCFs (GATK): Yes
* RunMutect1 (GATK): Yes
* RunMutect2 (GATK): Yes
* RunFreeBayes (freebayes): Yes, (but [not documented](https://github.com/ekg/freebayes/issues/90))
* RunStrelka (Strelka): Yes
* RunSingleStrelka (Strelka): Yes
* RunManta (Manta): Yes
* RunSingleManta (Manta): Yes
* RunAlleleCount: I don’t know

In summary:
* qualimap probably does not support CRAM natively
* Someone else needs to check RunAlleleCount

**Comment by MaxUlysse** [comment]
UPPMAX has made a guide about this isssue: https://www.uppmax.uu.se/support/user-guides/using-cram-to-compress-bam-files/

**Comment by marcelm** [comment]
Additionally, I believe that GATK, Manta, FreeBayes, Picard, `samtools stats` and probably Strelka accept CRAM files via the same command-line options that are currently used to provide BAM input. This would mean that theoretically these processes do not need to be changed at all. However, there are some hard-coded `.bam` strings in there, which needs to be updated.

Slightly bigger changes are required for MapReads and MergeBams.

**Comment by MaxUlysse** [comment]
![](https://github.trello.services/images/mini-trello-icon.png) [Add support for CRAM](https://trello.com/c/0YagURh4/48-add-support-for-cram)

**Comment by MaxUlysse** [comment]
cf https://github.com/nf-core/ExoSeq/issues/14

**Comment by MaxUlysse** [comment]
Transferred to [nf-core/sarek](https://github.com/nf-core/sarek/)
